fenix/app/src/main/java/org/mozilla/fenix/shopping/middleware/ProductAnalysisMapper.kt

```diff
@@ -29,6 +29,10 @@ private fun ProductAnalysis.toProductReview(): ProductReviewState =
         }
     } else if (notEnoughReviews && !needsAnalysis) {
         ProductReviewState.Error.NotEnoughReviews
+    } else if (deletedProduct) {
```

Comment:
Can this be true and `notEnoughReviews` is also true? If so, in that case, this'll map to `NotEnoughReviews`, should that happen or `NotAvailable`?

Reply:
Based on this [comment](https://github.com/mozilla-mobile/firefox-android/pull/4756/files#r1422911249), we should also check for when `deletedProductReported` is true. The above question/clarification applies for this also.

Reply:
O good point, moved up the `else if (deletedProduct)` before checking `notEnoughReviews` and also added check for `deletedProductReported`

Reply:
👌